src/Main/FreeCADGuiPy.cpp

```diff
@@ -305,6 +305,12 @@ QWidget* setupMainWindow()
             mw->setWindowTitle(QString::fromLatin1(App::Application::Config()["ExeName"].c_str()));
         }
 
+        // set toolbar icon size
+        ParameterGrp::handle hGrp = Gui::WindowParameter::getDefaultParameter()->GetGroup("General");
+        int size = hGrp->GetInt("ToolbarIconSize", 0);
+        if (size >= 16) // must not be lower than this

```

Comment:
And what happens if not?

Reply:
Hi @benj5378,
I guess you are referring to the comment in the last line: "must not be lower than this".
Good question -- I also asked this myself. As I wrote in my third sentence, the code is just copied from another place in this repo. So let's have a look at this original code with exactly this comment.
Traversing the blame layers in this FreeCAD repository does not bring us back to the author because the history stops suddenly. But there is an older archive, and there we find it:
`https://github.com/berndhahnebach/All_FreeCAD/commit/8077736ad6eb4b17d034ac9474ea58bc68bcadb1`
So if you want to know the answer, you can ask wwmayer who authored this in 2011.
(I did not make the url a link and wwmayer a mention to avoid unnecessary messages.)

Reply:
> And what happens if not?

If not then the icon size of the main window won't be overridden but the default value (24x24 px) will be kept.